### ISSUE TITLE
libcloud could not upload empty files to azure storage account

### ISSUE THREAD

**Issue by WentingWu666666** [thelastpickle/cassandra-medusa#533]
[Project board link](https://github.com/orgs/k8ssandra/projects/8/views/1?pane=issue&itemId=13951970)

SSTable-Attached Secondary Index (SASI) files (SI_*.db) could be empty. Medusa uses azure storage driver from libcloud to upload small files and az cli command to upload large files. However, libcloud fails to upload empty files with 403 error. Thus, we could use az cli command to upload empty files.

┆Issue is synchronized with this [Jira Task](https://k8ssandra.atlassian.net/browse/K8SSAND-1823) by [Unito](https://www.unito.io)
┆friendlyId: K8SSAND-1823
┆priority: Medium

**Comment by rzvoncek** [comment]
Hi, we recently overhauled the storage backends, and this probably got fixed. I'll have a look if it's still an issue.

**Comment by rzvoncek** [comment]
So it seems this has been resolved. I've added some tests to cover this, and after running them, I checked our IT buckets and saw the empty files everywhere:

S3:
![Screenshot 2023-09-21 at 16 49 58](https://github.com/thelastpickle/cassandra-medusa/assets/2589903/6e4d4778-8bb8-468a-b02d-9aa5a64b411a)

GCP:
![Screenshot 2023-09-21 at 16 52 58](https://github.com/thelastpickle/cassandra-medusa/assets/2589903/8da1f0af-25b5-4112-8826-c0a03b57ed87)

Azure:
![Screenshot 2023-09-21 at 16 54 32](https://github.com/thelastpickle/cassandra-medusa/assets/2589903/25676544-f8da-4f7d-9abd-0c136da1ae0f)

We will not see the empty blob when listing blobs, because of #595.

However, we do download the empty files because they get recorded in the manifest.
